packages/builder/src/components/backend/DataTable/TableDataTable.svelte

```diff
@@ -46,10 +47,7 @@
     tables.replaceTable(id, e.detail)
 
     // We need to refresh datasources when an external table changes.
-    // Type "external" may exist - sometimes type is "table" and sometimes it
-    // is "external" - it has different meanings in different endpoints.
-    // If we check both these then we hopefully catch all external tables.
-    if (e.detail?.type === "external" || e.detail?.sql) {
+    if (e.detail?.sourceType === DB_TYPE_EXTERNAL || e.detail?.sql) {
```

Comment:
We can remove the other part of this statement now (`|| e.detail?.sql)`) since we know that sourceType = external will always catch all external tables. Yay!